src/overlay005/save_info_window.c

```diff
@@ -0,0 +1,225 @@
+
+    int maxDigits, padding;
+
+    if (saveInfo->pokedexCount >= 100) {
+        maxDigits = 3;
+        padding = PADDING_MODE_NONE;
+    } else if (saveInfo->pokedexCount >= 10) {
+        maxDigits = 3;
+        padding = PADDING_MODE_SPACES;
+    } else {
+        maxDigits = 2;
+        padding = PADDING_MODE_SPACES;
+    }
+
+    StringTemplate_SetNumber(strTemplate, 3, saveInfo->pokedexCount, maxDigits, padding, CHARSET_MODE_EN);
+
+    int playTimeHours = PlayTime_GetHours(saveInfo->playTime);
+
+    if (playTimeHours >= 100) {
+        maxDigits = 3;
+        padding = PADDING_MODE_NONE;
+    } else if (playTimeHours >= 10) {
+        maxDigits = 3;
+        padding = PADDING_MODE_SPACES;
+    } else {
+        maxDigits = 2;
+        padding = PADDING_MODE_SPACES;
+    }
+
+    StringTemplate_SetNumber(strTemplate, 4, playTimeHours, maxDigits, padding, CHARSET_MODE_EN);
+    StringTemplate_SetNumber(strTemplate, 5, PlayTime_GetMinutes(saveInfo->playTime), 2, PADDING_MODE_ZEROES, CHARSET_MODE_EN);
+}
+
+static int SaveInfoWindow_Height(const SaveInfo *saveInfo)
+{
+    if (saveInfo->pokedexCount != 0) {
+        return SAVE_INFO_WINDOW_HEIGHT;
+    } else {
+        return SAVE_INFO_WINDOW_HEIGHT - 2;
+    }
+}
+
+static void SaveInfoWindow_PrintText(const SaveInfoWindow *saveInfoWin)
+{
+    int fontSpacing = Font_GetAttribute(FONT_SYSTEM, FONTATTR_MAX_LETTER_HEIGHT) + Font_GetAttribute(FONT_SYSTEM, FONTATTR_LINE_SPACING);
+    int yOffset = 0;
+    Strbuf *buf = MessageUtil_ExpandedStrbuf(saveInfoWin->strTemplate, saveInfoWin->msgLoader, sSaveInfoLabels[0], saveInfoWin->heapID);
+
+    Text_AddPrinterWithParams(saveInfoWin->window, FONT_SYSTEM, buf, 0, yOffset, TEXT_SPEED_NO_TRANSFER, NULL);
+    Strbuf_Free(buf);
+
+    int xOffset;
+
+    for (int i = 1; i < NELEMS(sSaveInfoLabels); i++) {
+        if (sSaveInfoLabels[i] == SAVE_INFO_LABEL_POKEDEX_COUNT && saveInfoWin->saveInfo.pokedexCount == 0) {
+            continue;
+        }
+
+        yOffset += fontSpacing;
+        buf = MessageLoader_GetNewStrbuf(saveInfoWin->msgLoader, sSaveInfoLabels[i]);
+
+        Text_AddPrinterWithParams(saveInfoWin->window, FONT_SYSTEM, buf, 0, yOffset, TEXT_SPEED_NO_TRANSFER, NULL);
+        Strbuf_Free(buf);
+
+        buf = MessageUtil_ExpandedStrbuf(saveInfoWin->strTemplate, saveInfoWin->msgLoader, sSaveInfoValues[i - 1], saveInfoWin->heapID);
+        xOffset = SAVE_INFO_WINDOW_WIDTH * 8 - Font_CalcStrbufWidth(FONT_SYSTEM, buf, Font_GetAttribute(FONT_SYSTEM, FONTATTR_LETTER_SPACING));
+
+        Text_AddPrinterWithParams(saveInfoWin->window, FONT_SYSTEM, buf, xOffset, yOffset, TEXT_SPEED_NO_TRANSFER, NULL);
+        Strbuf_Free(buf);
+    }
+}
+
+void SaveInfoWindow_NewWindow(SaveInfoWindow *saveInfoWin)
+{
+    saveInfoWin->window = Heap_AllocFromHeap(saveInfoWin->heapID, sizeof(Window));
+
+    Window_Add(saveInfoWin->bgConfig, saveInfoWin->window, saveInfoWin->bgLayer, 1, 1, saveInfoWin->width, saveInfoWin->height, 13, 393);
```

Comment:
**question**: `393` seems kinda magick-y to me; is there a set of constants that this is based off of?

This should also probably be in hex to signify that it's an address (since it's a base tile offset).